legate/core/machine.py

```diff
@@ -168,6 +168,11 @@ def __eq__(self, other: object) -> bool:
                 return False
         return True
 
+    @property
+    def num_procs(self) -> int:
```

Comment:
Just curious, any reason that you find `machine.num_procs` better than `len(machine)`? Both would compute the same value.